packages/ra-core/src/i18n/useTranslateLabel.spec.tsx

```diff
@@ -0,0 +1,155 @@
+        expect(screen.queryByText(/title/)).toBeNull();
+    });
+
+    it('should return the label element when provided', () => {
+        render(
+            <TestTranslationProvider>
+                <TranslateLabel
+                    label={<span>My title</span>}
+                    source="title"
+                    resource="posts"
+                />
+            </TestTranslationProvider>
+        );
+        screen.getByText('My title');
+    });
+
+    it('should return the label text when provided', () => {
+        render(
+            <TestTranslationProvider messages={{}}>
+                <TranslateLabel
+                    label="My title"
+                    source="title"
+                    resource="posts"
+                />
+            </TestTranslationProvider>
+        );
+        screen.getByText('My title');
+    });
+
+    it('should return the translated label text when provided', () => {
+        render(
+            <TestTranslationProvider
+                messages={{
+                    test: { title: 'My title' },
+                }}
+            >
+                <TranslateLabel
+                    label="test.title"
+                    source="title"
+                    resource="posts"
+                />
+            </TestTranslationProvider>
+        );
+        screen.getByText('My title');
+    });
+
+    it('should return the inferred label from source and resource when no label is provided', () => {
+        render(
+            <TestTranslationProvider messages={{}}>
+                <TranslateLabel source="title" resource="posts" />
+            </TestTranslationProvider>
+        );
+        screen.getByText('Title');
+    });
+
+    it('should return the translated inferred label from source and resource when no label is provided', () => {
+        render(
+            <TestTranslationProvider
+                messages={{
+                    resources: {
+                        posts: {
+                            fields: {
+                                title: 'My Title',
+                            },
+                        },
+                    },
+                }}
+            >
+                <TranslateLabel source="title" resource="posts" />
+            </TestTranslationProvider>
+        );
+        screen.getByText('My Title');
+    });
+
+    it('should return the inferred label from SourceContext when no label is provided but a SourceContext is present', () => {
+        render(
+            <TestTranslationProvider messages={{}}>
+                <SourceContextProvider
+                    value={{
+                        getSource: source => source,
+                        getLabel: source => `test.${source}`,
+                    }}
+                >
+                    <TranslateLabel source="title" resource="posts" />
+                </SourceContextProvider>
+            </TestTranslationProvider>
+        );
+        screen.getByText('Title');
```

Comment:
This I don't understand, as the labelFrom Source Context will be `test.title`

Reply:
Because the translation does not exist so we fall back to inferred label.